libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/SFSwiftDetectUtil.m

```diff
@@ -29,23 +29,23 @@
 @implementation SFSwiftDetectUtil
 
 +(BOOL)isSwiftApp{
-
-    /* 
+    
+    /*
      Detects swift app based on Swift name mangling pattern:
      See: https://en.wikipedia.org/wiki/Name_mangling#Swift
-    */
-    if([[NSThread callStackSymbols] count]>2){
-        NSArray *components = [[NSThread callStackSymbols][1] componentsSeparatedByCharactersInSet:[NSCharacterSet whitespaceCharacterSet]];
-        components = [components filteredArrayUsingPredicate:[NSPredicate predicateWithFormat:@"SELF <> ''"]];
-        
-        if([components count]>5){
-            NSString *callerName = components[[components count] - 3];
-            if(callerName && [callerName hasPrefix:kSFSwiftPrefix]){
+     */
+    for (NSString* callStackSymbol in [NSThread callStackSymbols]) {
```

Comment:
Checking the whole stack - the way it was written before, it would only work if called directly from swift code.